src/cloud_provider/digitalocean/common.rs

```diff
@@ -23,21 +28,41 @@ pub fn kubernetes_config_path(
         workspace_directory, kubernetes_cluster_id
     );
 
-    let kubeconfig = download_space_object(
-        spaces_access_id,
-        spaces_secret_key,
-        kubernetes_config_bucket_name.as_str(),
-        kubernetes_config_object_key.as_str(),
-        region,
-    );
-    match kubeconfig {
-        Ok(body) => {
+    let result = retry::retry(Fixed::from_millis(3000).take(5), || {
```

Comment:
shouldn't we add a sleep between each try?

Reply:
nop

Reply:
it's due a call that never give the control back ... will work on it on the futur PR.